Azure/sonic-platform-common
# sfp: add initial support for CMIS

Signed-off-by: Dante Su <dante.su@broadcom.com>



#### Why I did it

Add initial support for CMIS application initialization and loopback

#### How I did it

1. sonic-buildimage: Use Quanta IX9 as a reference design
2. sonic-platform-common: Update inf8628.py and qsfp_dd.py for the common QSFP-DD parser library, sfp_base.py for the abstraction layer, and add sfp_standard.py as the common based class
3. sonic-platform-daemon: Add a CMIS manager process to parallelize the CMIS application initialization
4. sonic-utilities: Update sfputil and sfpshow to support CMIS application advertisement, and loopback

#### How to verify it

1. Bring up Quanta IX9 with default configuration and CMIS 4.0/5.0 QSFPDD modules
2. Issue 'show logging xcvrd | grep CMIS' to check if the application initialization succeeded
3. Issue 'show interfaces transceiver eeprom --dom' to verify both the static and DOM information parsers
4. Issue 'sudo sfputil show eeprom --dom' to verify both the static and DOM information parsers
5. Issue 'sudo sfputil loopback set mo,mi,ho,hi' to verify enabling the lookbacks (media-output/media-input/host-output/host-input)
6. Issue 'sudo sfputil loopback clr mo,mi,ho,hi' to verify disabling the lookbacks (media-output/media-input/host-output/host-input)
7. Issue 'sudo sfputil show loopback' to display the loopback settings

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog

<table>
<tr>
<th>Repo</th>
<th>PR</th>
</tr>
<tr>
<td>sonic-buildimage</td>
<td>https://github.com/Azure/sonic-buildimage/pull/8899</td>
</tr>
</table>

#### A picture of a cute animal (not mandatory but encouraged)
